tests/acceptance/guid-file/registration-file-detail-test.ts

```diff
@@ -98,7 +112,167 @@ module('Acceptance | guid file | registration files', hooks => {
+                // Funder name
+                assert.dom('[data-test-target-funder-name-label]').hasText('Funder',
+                    'File metadata funder text is properly set.');
+                assert.dom('[data-test-target-funder-name]').exists();
+                assert.dom(`[data-test-target-funder-name="${funder.funder_name}"]`)
+                    .containsText(funder.funder_name, `Funder name is unchanged for ${funder.funder_name}`);
+                // Award title
+                assert.dom('[data-test-target-funder-award-title-label]').hasText('Award title',
+                    'File metadata award title text is properly set.');
+                assert.dom('[data-test-target-funder-award-title]').exists();
+                assert.dom(`[data-test-target-funder-award-title="${funder.award_title}"]`)
+                    .containsText(funder.award_title, `Funder award title is unchanged for ${funder.funder_name}`);
+                // Award URI
+                assert.dom('[data-test-target-funder-award-uri-label]').hasText('Award URI',
+                    'File metadata award URI text is properly set.');
+                assert.dom('[data-test-target-funder-award-uri]').exists();
+                assert.dom(`[ data-test-target-funder-award-uri="${funder.award_uri}"]`)
+                    .containsText(funder.award_uri, `Funder award URI is unchanged for ${funder.funder_name}`);
+                // Award number
+                assert.dom('[data-test-target-funder-award-number-label]').hasText('Award number',
+                    'File metadata award number text is properly set.');
+                assert.dom('[data-test-target-funder-award-number]').exists();
+                assert.dom(`[data-test-target-funder-award-number="${funder.award_number}"`)
+                    .containsText(funder.award_number, `Funder award number is unchanged for ${funder.funder_name}`);
+            }
+        }
+        // Target title
+        assert.dom('[data-test-target-title-label]').hasText('Title',
+            'File metadata target title text is properly set.');
+        assert.dom('[data-test-target-title]').exists();
+        // Target description
+        assert.dom('[data-test-target-description-label]').hasText('Description',
+            'File metadata target description text is properly set.');
+        assert.dom('[data-test-target-description]').exists();
+    });
+
+    // Save and cancel buttons with write permissions
+    test('save and cancel', async function(this: ThisTestContext, assert) {
+        await visit(`/--file/${this.file.id}`);
+        assert.equal(currentURL(), `/--file/${this.file.guid}`);
+
+        // Verify download
+        if (!this.registration.isAnonymous) {
+            assert.dom('[data-test-download-button]').exists();
+        }
 
-    // test for switching views
+        // Verify edit
+        if (this.file.target.currentUserPermissions.includes(Permission.Write)) {
```

Comment:
This if statement, also, is not appropriate. You should know what your current user's permissions are and set them to be the appropriate permissions so that this these assertions will run.

Reply:
Ok, this is removed.

Reply:
This is now a permission on the registration instead of a conditional in the test.